### ISSUE TITLE
output have wrong scale

### ISSUE THREAD

**Issue by ghost** [Wovchena/text-detection-fots.pytorch#6]
@Wovchena 
The detected output txt seems to have a wrong scale.

Including the results: [results.zip](https://github.com/Wovchena/text-detection-fots.pytorch/files/3612543/results.zip)
consist of jpg, txt, and the converted json

```
 python test.py --images-folder ./test --output-folder ./out --checkpoint ./model/best_checkpoint.pt
```

Result:
![result1](https://user-images.githubusercontent.com/49877135/64905993-e7218080-d6e8-11e9-92d0-7686525bab2f.jpg)

**Comment by Wovchena** [comment]
I guess this depends on how you solved https://github.com/Wovchena/text-detection-fots.pytorch/issues/5.

**Comment by ghost** [comment]
I was not able to train, I was only able to test the pretrained model that you shared.
- Notice that the original image only have 7 polygon boxes, now notice that the more you increase the `IN_OUT_RATIO` the bigger it scales up, but the more polygon labels it's getting.

Groundtruth
![groundtruth](https://user-images.githubusercontent.com/49877135/64975955-e17a9500-d8b8-11e9-85f1-a8dda43f2b46.jpg)

IN_OUT_RATIO = 2
![2](https://user-images.githubusercontent.com/49877135/64975706-54374080-d8b8-11e9-8b7f-aa1da1aadfc8.jpg)

IN_OUT_RATIO = 3
![3](https://user-images.githubusercontent.com/49877135/64975710-5a2d2180-d8b8-11e9-8b5d-1c8935b3f2c0.jpg)

IN_OUT_RATIO = 4
![4](https://user-images.githubusercontent.com/49877135/64975723-61542f80-d8b8-11e9-8516-8d68002b41f7.jpg)

IN_OUT_RATIO = 4.1
![41](https://user-images.githubusercontent.com/49877135/64975753-687b3d80-d8b8-11e9-8210-013d8ca99c94.jpg)

**Comment by Wovchena** [comment]
`IN_OUT_RATIO = 4` looks correct (I mean that this result can be expected).
1) Is this a raw net's output?
2) Have you tried the model with other images?

If you run pretrained model, you better run it with this code https://github.com/Wovchena/text-detection-fots.pytorch/pull/1/commits/90db90cb915d7d611a7da4380b2cca3add6b9d8b, where the model was originally trained and evaluated.

**Comment by ghost** [comment]
@Wovchena Yup,
master at [90db90cb](https://github.com/Wovchena/text-detection-fots.pytorch/tree/90db90cb915d7d611a7da4380b2cca3add6b9d8b) both training and detection work.

Your question:
1. These are the raw detected polygons converted to `.json` and opened at `labelme`.

Conclusion:
- So, what do you think caused the **detection** issue?
- Will you fix the **training data-loader**? Because it's seems that it is not loading the data properly.


Below are proof that master [90db90cb](https://github.com/Wovchena/text-detection-fots.pytorch/tree/90db90cb915d7d611a7da4380b2cca3add6b9d8b) is working:
![00](https://user-images.githubusercontent.com/49877135/64983735-e431b600-d8c9-11e9-8544-849d6e72a53b.jpg)


![0](https://user-images.githubusercontent.com/49877135/64983393-0b3bb800-d8c9-11e9-9376-f5f0cbd5dc45.jpg)

**Comment by ghost** [comment]
Another thing,I have trained for 6 hours using 90db90cb, the model still doesn't recognize anything, eventhough the loss is low.

**Comment by Wovchena** [comment]
> So, what do you think caused the detection issue?

What do you mean by "issue"? You agreed that it works for 90db90cb. As for running the prtrained net on `master`, well, since the model was trained, there were changes affecting its architecture and postprocessing.

> Will you fix the training data-loader? Because it's seems that it is not loading the data properly.

What are the reasons to think that it does not load the data properly? Any way I paused the work on this repo for a while, so the answer is not soon.

> Another thing,I have trained for 6 hours using 90db90c, the model still doesn't recognize anything, eventhough the loss is low.

What's the value for loss? As I remember, it should be smaller than 1.0. You can decrease threshold for postprocessing, because it was adapted for a trained net (0.95). So if the model did not complete its training, it will not give you such high confidences.

**Comment by ghost** [comment]
@Wovchena but the model that i trained doesn't recognize anything

**Comment by Wovchena** [comment]
> What's the value for loss? As I remember, it should be smaller than 1.0. You can decrease threshold for postprocessing, because it was adapted for a trained net (0.95). So if the model did not complete its training, it will not give you such high confidences.

**Comment by Wang-Zhenxing** [comment]
Can you get this result "The best result "recall": 0.7746750120365913, "precision": 0.8085427135678392, "hmean": 0.7912466191295796" which comes from the pretrained models provided by author?  I got the result that is "Calculated!{"recall": 0.0, "precision": 0.0, "hmean": 0, "AP": 0}"! 
But I trained a model using the ICDAR2015 train-set in total 100-epochs, Mean loss=0.154, and it got the result is "Calculated!{"recall": 0.0014443909484833895, "precision": 0.0017482517482517483, "hmean": 0.0015818613234906405, "AP": 0}" !!
Can you help me solve this problem? Thank you.

**Comment by Wovchena** [comment]
@Srwzx, if you run pretrained model, you better run it with this code 90db90c, where the model was originally trained and evaluated.
As for your training, I still didn't fix `master` to use correct scale (assuming you took last commit from `master`). Did you take https://github.com/Wovchena/text-detection-fots.pytorch/issues/5#issuecomment-525807027 into account?

**Comment by Wang-Zhenxing** [comment]
Oh, sorry, I haven't take #5 (comment) into account! So, you mean I should replace IN_OUT_RATIO = 2 with IN_OUT_RATIO = 4 both in datasets.py and modules/parse_polys.py? 
And I change the code back to the original code for pretrain model, which I can obtain the result -- "The best result "recall": 0.7746750120365913, "precision": 0.8085427135678392, "hmean": 0.7912466191295796"??

**Comment by Wovchena** [comment]
> So, you mean I should replace IN_OUT_RATIO = 2 with IN_OUT_RATIO = 4 both in datasets.py and modules/parse_polys.py?

Correct.

I'm not sure, but I think I have introduced that scale bug after 90db90c. So you have two ways:
1. Train your own model just on 90db90c
2. Fix scale bug in master and train your model on fixed master.

> And I change the code back to the original code for pretrain model, which I can obtain the result -- "The best result "recall": 0.7746750120365913, "precision": 0.8085427135678392, "hmean": 0.7912466191295796"??

I'm not sure I got the question. To obtain the same result for pretrained model you should change code back to the original code.

**Comment by Srwzx** [comment]
Okay, thanks.  I will try to modified this code to original code.  If I have some questions, I will email you. Thank you very much.------------------ 原始邮件 ------------------
发件人: "Zlobin Vladimir"<notifications@github.com>
发送时间: 2019年11月8日(星期五) 晚上11:08
收件人: "Wovchena/text-detection-fots.pytorch"<text-detection-fots.pytorch@noreply.github.com>;
抄送: "zxwang"<1002446431@qq.com>;"Mention"<mention@noreply.github.com>;
主题: Re: [Wovchena/text-detection-fots.pytorch] output have wrong scale(#6)


 
So, you mean I should replace IN_OUT_RATIO = 2 with IN_OUT_RATIO = 4 both in datasets.py and modules/parse_polys.py?
  
Correct.
 
I'm not sure, but I think I have introduced that scale bug after 90db90c. So you have two ways:
  
Train your own model just on 90db90c
 
Fix scale bug in master and train your model on fixed master.
   
And I change the code back to the original code for pretrain model, which I can obtain the result -- "The best result "recall": 0.7746750120365913, "precision": 0.8085427135678392, "hmean": 0.7912466191295796"??
  
I'm not sure I got the question. To obtain the same result for pretrained model you should change code back to the original code.
 
&mdash;

**Comment by Wovchena** [comment]
Fixed by 816fc8b201ea35be83f57457f42e23067e65a22f.

**Comment by QuickLearner171998** [comment]
> I was not able to train, I was only able to test the pretrained model that you shared.
> 
> * Notice that the original image only have 7 polygon boxes, now notice that the more you increase the `IN_OUT_RATIO` the bigger it scales up, but the more polygon labels it's getting.
> 
> Groundtruth
> ![groundtruth](https://user-images.githubusercontent.com/49877135/64975955-e17a9500-d8b8-11e9-85f1-a8dda43f2b46.jpg)
> 
> IN_OUT_RATIO = 2
> ![2](https://user-images.githubusercontent.com/49877135/64975706-54374080-d8b8-11e9-8b7f-aa1da1aadfc8.jpg)
> 
> IN_OUT_RATIO = 3
> ![3](https://user-images.githubusercontent.com/49877135/64975710-5a2d2180-d8b8-11e9-8b5d-1c8935b3f2c0.jpg)
> 
> IN_OUT_RATIO = 4
> ![4](https://user-images.githubusercontent.com/49877135/64975723-61542f80-d8b8-11e9-8516-8d68002b41f7.jpg)
> 
> IN_OUT_RATIO = 4.1
> ![41](https://user-images.githubusercontent.com/49877135/64975753-687b3d80-d8b8-11e9-8210-013d8ca99c94.jpg)

How did u visualize the raw net outputs?
